crates/types/src/config/common.rs

```diff
@@ -81,19 +82,27 @@ pub struct CommonOptions {
     /// If set, the node insists on acquiring this node ID.
     pub force_node_id: Option<PlainNodeId>,
 
-    /// # Cluster Name
+    /// # Cluster name
     ///
     /// A unique identifier for the cluster. All nodes in the same cluster should
     /// have the same.
     cluster_name: String,
 
-    /// If true, then a new cluster is bootstrapped. This node *must* have an admin
-    /// role and a new nodes configuration will be created that includes this node.
+    /// # Auto cluster provisioning
     ///
-    /// Bootstrap is allowed by default, unless it is explicitly disabled in config files.
+    /// If true, then this node is allowed to automatically provision as a new cluster.
+    /// This node *must* have an admin role and a new nodes configuration will be created that includes this node.
+    ///
+    /// auto-provision is allowed by default in development mode and is disabled if restate-server runs with `--production` flag
+    /// to prevent cluster nodes from forming their own clusters, rather than forming a single cluster.
+    ///
+    /// Use `restatectl` to provision the cluster/node if automatic provisioning is disabled.
+    ///
+    /// This can also be explicitly disabled by setting this value to false.
     ///
     /// Default: true
-    pub allow_bootstrap: bool,
+    #[serde(alias = "allow-bootstrap")]
```

Comment:
The alias might actually result into duplicate fields when trying to read a configuration where `allow-bootstrap` is defined.

Reply:
Ugh, I couldn't actually find a good solution to this :( without investing too much time, and I don't want us to give up on the rename.

I'm not sure if there are good ideas here :(

Reply:
The way I solved it in another PR is by introducing a Shadow variant: https://github.com/restatedev/restate/pull/2571/files#diff-b07de33428bbafda6c26f429c0f36178f5ef8e6215e7a1f187c72142417ecc29R249. Not particularly proud of it, though.